### ISSUE TITLE
Not asking for run time permission and its directly showing gps provider popup

### ISSUE THREAD

**Issue by JagadishaIncture** [the-best-is-best/klocation#3]
val viewModel by remember { mutableStateOf(mapViewModel) }

    if (viewModel.requestPermission) {
        KLocationService().EnableLocation()
        viewModel.requestPermission = false
    }

**Comment by the-best-is-best** [comment]
Ios need to add permission in info.plist

**Comment by the-best-is-best** [comment]
```
<key>NSLocationWhenInUseUsageDescription</key>
<string>We need your location to provide better services when the app is in use.</string>

```

**Comment by JagadishaIncture** [comment]
in android its not working

**Comment by the-best-is-best** [comment]
(viewModel.requestPermission)

Is it be true or not

**Comment by the-best-is-best** [comment]
![Screenshot_2024-12-10-14-39-00-021_com.google.android.permissioncontroller.jpg](https://github.com/user-attachments/assets/3a23a0bb-b751-40bf-9c8a-524866865e66)

**Comment by JagadishaIncture** [comment]
var requestPermission by mutableStateOf(false)
 in viewmodel

**Comment by the-best-is-best** [comment]
Yes you will ask for GPS first and need to call it again to ask for permission.


Because you can't ask permission without GPS Enabled in ios

**Comment by JagadishaIncture** [comment]
But this code is not working in Android for me.. no runtime permission request

**Comment by the-best-is-best** [comment]
Check permission in android and ios
